docker image for ledger-on-memory (#5256)
digital-asset/daml: ledger/ledger-on-memory/README.md

# Overview
This document is to help internal engineers work with the ledger-on-memory.
Ledger on memory is a kv-based ledger that uses a simple in-memory map as an underlying

> storage, it uses either H2 or Postgres as its index db.

Reviewer: ```suggestion

This document is to help internal engineers work with `ledger-on-memory`.

`ledger-on-memory` is a key/value-based ledger that uses a simple in-memory map as an
underlying storage. It uses either H2 or PostgreSQL as its index database.
```

> # Ledger On Memory

Reviewer: Probably worth mentioning you can also run `bazel run //ledger/ledger-on-memory:app`.

Author: Done

To build a fat JAR with the server built from HEAD run

    bazel build //ledger/ledger-on-memory:app_deploy.jar

> ledger-on-memory application can be run from command line with the following command:

Reviewer: ```suggestion
The application can be run from command line with the following command:
```

    java -Xmx4G -XX:+UseG1GC -jar bazel-bin/ledger/ledger-on-memory/app_deploy.jar --participant participant-id=foo,port=6865

as run from the main project root directory (adjust the location of the JAR according to 

> your working directory).

Reviewer: ```suggestion
As run from the main project root directory (adjust the location of the JAR according to 
your working directory).
```

## Docker

> Ledger on memory can be put into a docker container by first building it

Reviewer: ```suggestion
Ledger On Memory can be put into a Docker container by first building it:
```

    bazel build ledger/ledger-on-memory:app-image

> Then by running the bazel target that registers it in the local docker repository

Reviewer: ```suggestion
Then by running the Bazel target that imports the image into your local Docker repository, and then
runs a container from that image.
```

    bazel run ledger/ledger-on-memory:app-image
    
The container will be registered under a name `bazel/ledger/ledger-on-memory`.

> It can be then run using docker:

Reviewer: ```suggestion
The container will fail to run, but the image has now been loaded into your local Docker
repository under the name `bazel/ledger/ledger-on-memory`.

It can then be run using Docker:
```

    docker run \
        -it \
        --rm \
        --name ledger
        --network docker_default \
        -p 6861:6865 
        bazel/ledger/ledger-on-memory:app-image \
            --participant participant-id=blah,port=6865,address=0.0.0.0
            
 Above command maked the ledger available to clients from the host machine on the 
 `localhost:6861`. It is also available to other docker containers on the `docker_default`

>  network under `ledger:6865`.

Reviewer: ```suggestion
The above command makes the ledger available to clients from the host machine at the address 
`localhost:6861`. It is also available to other docker containers on the `docker_default`
network under `ledger:6865`.
```

underlying storage. It uses either H2 or PostgreSQL as its index database.

# Ledger On Memory

To build a fat JAR with the server built from HEAD run

    bazel build //ledger/ledger-on-memory:app_deploy.jar

The application can be run from command line with the following command:

    java -Xmx4G -XX:+UseG1GC -jar bazel-bin/ledger/ledger-on-memory/app_deploy.jar --participant participant-id=foo,port=6865

As run from the main project root directory (adjust the location of the JAR according to 
your working directory).

## Docker

Ledger On Memory can be put into a Docker container by first building it:

    bazel build ledger/ledger-on-memory:app-image

Then by running the Bazel target that imports the image into your local Docker repository, and then
runs a container from that image.

    bazel run ledger/ledger-on-memory:app-image
    
The container will fail to run, but the image has now been loaded into your local Docker
repository under the name `bazel/ledger/ledger-on-memory`.

It can then be run using Docker:

    docker run \
        -it \
        --rm \
        --name ledger
        --network docker_default \
        -p 6861:6865 

>         bazel/ledger/ledger-on-memory:app-image \

Reviewer 2: Shouldn’t this be `bazel-bin` instead of `bazel`?

Reviewer: Nope, it's not a filesystem path, it's a Docker image tag. The Bazel rule uses your Bazel target path but prefixes "bazel/" to it.

Reviewer 2: Oh of course, thanks for the explanation!

# Overview

This document is to help internal engineers work with `ledger-on-memory`.

`ledger-on-memory` is a key/value-based ledger that uses a simple in-memory map as an
underlying storage. It uses either H2 or PostgreSQL as its index database.

# Ledger On Memory

To build a fat JAR with the server built from HEAD run

    bazel build //ledger/ledger-on-memory:app_deploy.jar

The application can be run from command line with the following command:

    java -Xmx4G -XX:+UseG1GC -jar bazel-bin/ledger/ledger-on-memory/app_deploy.jar --participant participant-id=foo,port=6861

As run from the main project root directory (adjust the location of the JAR according to 
your working directory).

> Alternatively, the application can be run using the bazel command

Reviewer: ```suggestion
Alternatively, the application can be run using the Bazel command:
```
